homeassistant/components/air_quality/ampio.py

```diff
@@ -0,0 +1,113 @@
+"""
+Support for Ampio Air Quality data.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/air_quality/ampio/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.air_quality import (
+    PLATFORM_SCHEMA, AirQualityEntity)
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['asmog==0.0.6']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Data provided by Ampio'
+CONF_STATION_ID = 'station_id'
+SCAN_INTERVAL = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_STATION_ID): cv.string,
+
+    vol.Optional(CONF_NAME): cv.string,
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the Ampio Smog air quality platform."""
+    from asmog import AmpioSmog
+
+    name = config.get(CONF_NAME)
+    station_id = config[CONF_STATION_ID]
+
+    session = async_get_clientsession(hass)
+    api = AmpioSmogMapData(AmpioSmog(station_id, hass.loop, session))
+
+    await api.async_update()
+
+    if not api.api.data:
+        _LOGGER.error("Station %s is not available", station_id)
+        return
+
+    async_add_entities([AmpioSmogQuality(api, station_id, name)], True)
+
+
+class AmpioSmogQuality(AirQualityEntity):
+    """Implementation of an Ampio Smog air quality entity."""
+
+    def __init__(self, api, station_id, name):
+        """Initialize the air quality entity."""
+        self._ampio = api
+        self._station_id = station_id
+        self._name = name or api.api.name
+
+    @property
+    def name(self):
+        """Return the name of the air quality entity."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return unique_name."""
+        return "ampio_smog_{}".format(self._station_id)
+
+    @property
+    def should_poll(self):
+        """Pool for new date update."""
+        return True
+
+    @property
+    def particulate_matter_2_5(self):
+        """Return the particulate matter 2.5 level."""
+        return self._ampio.api.pm2_5
+
+    @property
+    def particulate_matter_10(self):
+        """Return the particulate matter 10 level."""
+        return self._ampio.api.pm10
+
+    @property
+    def unit_of_measurement(self):
```

Comment:
This is not part of AirQualityEntity. Remove it.

Reply:
It is inherited from Entity, isn't it? Why should I remove it, then. With unit, it is nicely presented in GUI.

Reply:
Because we can't convert it. We are running into the same issue as with the `weather` implementation as long as [Unit systems conversion](https://github.com/home-assistant/architecture/issues/10) is not done. The `air_quality` frontend card will need to handle the unit of measurements for now.

Reply:
I read the [Unit systems conversion](https://github.com/home-assistant/architecture/issues/10) thread and one thing is not clear to me. If we can't convert the units automatically, shouldn't we have a let's call it a base unit of measurement for each sensor? If my sensor always reports temp in C and I set the unit_system to imperial, how do we know that this particular value must be converted from C to F?

Reply:
Temperature units are converted automatically in the base entity class.